### PR TITLE
Rework cudf::strings::from_timestamps to use make_strings_children

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Rework `cudf::strings::from_timstamps` to use the internal `make_strings_children`. This simplifies the code and allows the operation to throw an error if it the output would exceed the size limit of a column.
No function has been added, removed, or changed.

Reference https://github.com/rapidsai/cudf/issues/12167

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/src/strings/convert/convert_datetime.cu [hunk 1]
```diff
@@ -747,34 +753,18 @@ struct from_timestamp_base {
 
     return result;
   }
-};
 
-template <typename T>
-struct from_timestamps_size_fn : public from_timestamp_base<T> {
```
Comment (davidwendt) [inline cpp/src/strings/convert/convert_datetime.cu:753]: Basically combining `from_timestamps_size_fn` and `datetime_formatter` functors into one functor: `datetime_formatter_fn`

File: cpp/src/strings/convert/convert_datetime.cu [hunk 2]
```diff
@@ -700,7 +706,7 @@ struct from_timestamp_base {
    *     modulo(-1,60) -> 59
    * @endcode
    */
-  [[nodiscard]] __device__ int32_t modulo_time(int64_t time, int64_t base) const
+  __device__ int32_t modulo_time(int64_t time, int64_t base) const
```
Comment (ttnghia) [inline cpp/src/strings/convert/convert_datetime.cu:709]: Why are you removing `[[nodiscard]]`?

File: cpp/src/strings/convert/convert_datetime.cu [hunk 3]
```diff
@@ -1057,6 +1021,19 @@ struct datetime_formatter : public from_timestamp_base<T> {
...
     }
   }
+
+  __device__ void operator()(size_type idx) const
+  {
+    if (d_timestamps.is_null(idx)) {
+      if (!d_chars) { d_offsets[idx] = 0; }
+      return;
+    }
+    auto tstamp = d_timestamps.element<T>(idx);
```
Comment (ttnghia) [inline cpp/src/strings/convert/convert_datetime.cu:1031]: `const`?

**Review by ttnghia — APPROVED** [review]

File: cpp/src/strings/convert/convert_datetime.cu [hunk 4]
```diff
@@ -1057,6 +1021,19 @@ struct datetime_formatter : public from_timestamp_base<T> {
...
+  {
+    if (d_timestamps.is_null(idx)) {
+      if (!d_chars) { d_offsets[idx] = 0; }
+      return;
+    }
+    auto const tstamp = d_timestamps.element<T>(idx);
+    if (d_chars)
+      timestamp_to_string(tstamp, d_chars + d_offsets[idx]);
+    else
+      d_offsets[idx] = compute_output_size(tstamp);
```
Comment (vuule) [inline cpp/src/strings/convert/convert_datetime.cu:1035]: I think blocks are preferred for all branches.
```suggestion
    if (d_chars) {
      timestamp_to_string(tstamp, d_chars + d_offsets[idx]);
    } else {
      d_offsets[idx] = compute_output_size(tstamp);
    }
```

**Comment by davidwendt** [comment]
@gpucibot merge

**Pull request merged**